core/o11y/src/macros.rs

```diff
@@ -6,38 +6,42 @@ macro_rules! handler_span_use_context {
 /// Use this at the beginning of `handle` methods to reduce the boilerplate.
 #[macro_export]
 macro_rules! handler_span {
-    ($target:expr, $actor:expr, $handler:expr, $msg:expr) => {{
-        let reflection_actor =
-            std::any::type_name::<Self>().rsplit_once("::").map_or("__", |(_, name)| name);
-        let span = tracing::debug_span!(
-            target: $target,
-            "handle",
-            handler = $handler,
-            actor = $actor,
-            reflection_actor = reflection_actor
-        )
-        .entered();
-
+    ($target:expr, $msg:expr) => {{
         let WithSpanContext { msg, context, .. } = $msg;
+
+        fn fix_name(name: &str) -> &str {
+            name.rsplit_once("::").map_or("_", |(_, name)| name)
+        }
+        fn type_name_of<T>(_: &T) -> &str {
+            fix_name(std::any::type_name::<T>())
+        }
+
+        let actor = fix_name(std::any::type_name::<Self>());
+        let handler = type_name_of(&msg);
+
+        let span = tracing::debug_span!(target: $target, "handle", handler, actor,).entered();
+
         span.set_parent(context);
         (span, msg)
     }};
 
     // Also specifies type of the received message.
-    ($target:expr, $actor:expr, $handler:expr, $msg:expr, $msg_type_fn:expr) => {{
+    ($target:expr, $msg:expr, $msg_type_fn:expr) => {{
```

Comment:
I’d change this to

```rust
($target:expr, $msg:expr, $($extra_fields:tt)*)
```

and then

```rust
let span = tracing::debug_span!(target: $target, "handle", handler, actor, $($extra_fields)*).entered();
```

That way callers can use the usual span KV syntax for their KVs, including the msg_type:

```rust
let msg_type = msg.msg.into();
handler_span!("client", msg, msg_type);
```